8827: migrate trial sessions to postgres (formerly flexion#10493)

TomElliottFlexion opened the pull request:
#8827

zacharybrown204 committed:
10493

Initial Commit
Created schema for trial session
Created mapper for trial session

zacharybrown204 committed:
10493

WIP
Moved seed data for trial sessions over to postgres
Created new persistance functions for postgres

favourustc committed:
10493: port over postgres getTrialSessionById to postgres and references

favourustc committed:
10493: WIP replace dynamo getTrialSessions with postgres version

zacharybrown204 committed:
10493

WIP
Moved over the rest of the dwTrialSession stuff, now need to create new tables for other related dynamo stuff

zacharybrown204 committed:
10493

WIP
Adding migration and mapping for TrialSessionWorkingCopy

zacharybrown204 committed:
10493

Seed data for trial session working copy
Updated a few interactors to use PG for trial session working copy
Merged two persistance functions

zacharybrown204 committed:
10493

WIP
Added table for paper pdfs in trial sessions.
Moved persistance functions for trial session working copy

zacharybrown204 committed:
10493

Refactoring create trial session to handle optional paper service pdfs
Type updates to the mapper
Removing paper pdfs from primary trial session table.

favourustc committed:
WIP replacing hearing with hearingId

TomElliottFlexion committed:
10493: add migrate note

zacharybrown204 committed:
Merge branch '10493' into 10493-trialsessions

zacharybrown204 committed:
10493

Update get trial session to grab pdf data from separate table.

zacharybrown204 committed:
10493

Updated references to updateTrialSession

zacharybrown204 committed:
10493

Moving trial session notification batch job over to postgres

zacharybrown204 committed:
10493

Typescript updates for trial notification processing

favourustc committed:
WIP Moving Case Hearing to Pg

zacharybrown204 committed:
10493

Updaing case hearing sql query

zacharybrown204 committed:
Merge remote-tracking branch 'origin/staging' into 10493-trialsessions

zacharybrown204 committed:
10493

Updating case hearing mapping query

zacharybrown204 committed:
10493

Fixing issues found in cypress tests #1

zacharybrown204 committed:
10493

Fixing issues found in cypress tests #2

zacharybrown204 committed:
10493

Added seed data that was missed in the conversion
Fixing a cypress test #3

zacharybrown204 committed:
10493

Updating trial session working copy parameters

TomElliottFlexion committed:
Merge branch 'staging' into 10493-trialsessions

Mwindo committed:
10493: adding an implicit wait to trial-session-filtering.cy.ts just to diagnose CI/CD failure we cannot re-create locally

DO NOT COMMIT TO STAGING!

Mwindo committed:
10493: adding an explicit wait in download-all-cases-from-trial-session.cy.ts

Mwindo committed:
10493: remove implicit wait that we were using to diagnose CI/CD failure; fix a few cerebral integration tests

Mwindo committed:
10493: add await to a few persistence functions to be consistent with other code

Mwindo committed:
10493: date to timestamptz in the migration file

TomElliottFlexion committed:
Merge branch 'staging' into 10493-trialsessions

TomElliottFlexion committed:
10493: temp workaround in getCasesByDocketNumbers to return dates correctly

TomElliottFlexion committed:
Merge branch '10493-trialsessions' of https://github.com/ustaxcourt/ef-cms into 10493-trialsessions

TomElliottFlexion committed:
10493: update erroneously passing test to remove case from a (not undefined) hearing

TomElliottFlexion committed:
10493: remember to commit all of the code

TomElliottFlexion committed:
Merge branch 'staging' into 10493-trialsessions

favourustc committed:
10493: camelcase NOTT in dismissedAlertForNott and hasNottBeenServed on trialsession

favourustc committed:
10493 updating trial hearing test

favourustc committed:
10493-updating add case to hearing

favourustc committed:
10493: fix typescript errors

favourustc committed:
10493: created mock file for trial sessions

favourustc committed:
10493: WIP modified trialsessions schema and migration

favourustc committed:
10493: WIP modified add case to trial session

favourustc committed:
WIP TCaseOrder persistence to new mapping table

favourustc committed:
WIP - update casse trial session

zacharybrown204 committed:
10493

Work on trial session - case mapping

favourustc committed:
10493: WIP modified add case to trial session

Mwindo commented on web-api/src/persistence/postgres/trialSessions/deleteTrialSession.ts:
```diff
@@ -0,0 +1,19 @@
+}: {
+  trialSessionId: string;
+}) => {
+  await pgDeleteFrom({
```
We can do an `await settlePromises` here

Mwindo commented on web-api/src/persistence/postgres/trialSessions/deleteCasesFromTrialSession.ts:
```diff
@@ -0,0 +1,19 @@
+import { pgDeleteFrom } from '../utils/operation/pgDeleteFrom';
+
+export const deleteCasesFromTrialSession = async ({
```
At some point when the dust settles, I feel like we'll need to figure out how to name things better. We have `deleteCasesFromTrialSession` and `removeCaseFromTrialSession`. These read as basically equivalent minus the plural/singular difference. In app code, they do slightly different things.

Mwindo commented on web-api/src/persistence/postgres/trialSessions/createOrUpdateTrialSessionCases.ts:
```diff
@@ -0,0 +1,35 @@
+
+import { toKyselyNewTrialSessionCase } from './mapper';
+
+export const createOrUpdateTrialSessionCases = async ({
```
Probably should just call this `upsertTrialSessionCases`.

TomElliottFlexion replied:
We _almost_ named it `upsertTrialSessionCases`

Mwindo commented on web-api/src/persistence/postgres/cases/getCasesByDocketNumbers.ts:
```diff
@@ -312,23 +314,46 @@ async function getCaseCorrespondenceByDocketNumber(docketNumbers: string[]) {
+  //     .execute(),
+  // );
+
+  // TODO 10493: this is a hack to make the the fromKyselyTrialSession functionality work; rethink this
```
My recommendation: unless this is leading to clear performance issues, put a pin in it for extra credit later, probably in a followup ticket. It is more important to have everything in Postgres and then refine than to try to get the details 100% right on the first pass.

Mwindo commented on web-api/src/business/useCases/trialSessions/setTrialSessionCalendarInteractor.ts:
```diff
@@ -134,22 +133,22 @@ export const setTrialSessionCalendarInteractor = async (
-    });
+      const caseOrderObject = trialSessionEntity.addCaseToCalendar(theCase);
+      return [theCase.validate().toRawObject(), caseOrderObject];
+    }) as [ExcludeMethods<Case>[], TCaseOrder[]];
```
`ExcludeMethods<Case>` can just be `RawCase` in this file

zacharybrown204 committed:
10493

Updating cases getting hearings

zacharybrown204 committed:
10493

Added trial session case seed data
Added workaround to deserialize dates in nested selects

zacharybrown204 committed:
10493

Fixing cypress tests #4?

zacharybrown204 committed:
10493

Knocking out todo items
Addressing failing tests #5? #6?

favourustc committed:
fixed script + update case & associations test

zacharybrown204 committed:
10493

Updating unit tests related to trial session PG migration

Mwindo committed:
10493: fix a typo I caused

zacharybrown204 committed:
10493

Unit test updates

zacharybrown204 committed:
10493 Updating unit tests

zacharybrown204 committed:
Merge staging into branch

zacharybrown204 committed:
10493 Updating unit tests

zacharybrown204 committed:
10493 Fixing shared unit tests

zacharybrown204 committed:
10493 Fixing typescript errors

zacharybrown204 committed:
10493 Fixing lint errors

zacharybrown204 committed:
10493 Fixing typescript error

TomElliottFlexion committed:
10493: resolve one-off type error

TomElliottFlexion committed:
Merge branch '10493-trialsessions' of https://github.com/ustaxcourt/ef-cms into 10493-trialsessions

TomElliottFlexion committed:
10493: note

TomElliottFlexion committed:
10493: WIP begin migration script, add script for checking diff between caseOrder and caseHearing

TomElliottFlexion committed:
10493: correct test script

TomElliottFlexion committed:
10493: make addedToSessionAt an optional field on case-trial session mapping record

TomElliottFlexion committed:
10493: continue work on data migration

TomElliottFlexion committed:
10493: note for testing

zacharybrown204 committed:
10493 Updating migration script

TomElliottFlexion committed:
Merge branch 'staging' of https://github.com/ustaxcourt/ef-cms into 10493-trialsessions

TomElliottFlexion committed:
Merge branch '10493-trialsessions' of https://github.com/ustaxcourt/ef-cms into 10493-trialsessions

TomElliottFlexion committed:
10493: update migration for accuracy and to avoid writing empty values to db, update mapper to stringify trialClerk object

TomElliottFlexion committed:
10493: update todo file

zacharybrown204 committed:
10493 Flattening jsonb columns in trial session table

TomElliottFlexion committed:
Merge branch 'staging' of https://github.com/ustaxcourt/ef-cms into 10493-trialsessions

TomElliottFlexion committed:
10493: update getTrialSession imports in new files

TomElliottFlexion committed:
Merge branch 'staging' of https://github.com/ustaxcourt/ef-cms into 10493-trialsessions

TomElliottFlexion committed:
Merge branch 'staging' of https://github.com/ustaxcourt/ef-cms into 10493-trialsessions

TomElliottFlexion committed:
10493: attempt to run trial session migration as part of deploy with new circle param

zacharybrown204 committed:
Fixing perofrmance issue when getting all tiral sessions.

zacharybrown204 committed:
Fixing an unaliased where condition in a joined query.

zacharybrown204 committed:
Adding distinct to get trial session queries. Was causing issues with duplicate records in tests.

zacharybrown204 committed:
10493

Updating mappers transforming trial session data. Removing out of date comments.

TomElliottFlexion committed:
10493: try using sql json_build_array() instead of js array

TomElliottFlexion committed:
Merge branch 'staging' of https://github.com/ustaxcourt/ef-cms into 10493-trialsessions

TomElliottFlexion committed:
10493: remove test script, unused dynamo types

TomElliottFlexion committed:
10493: remove commented code, comments

TomElliottFlexion committed:
10493: bring in fix from to-test branch

TomElliottFlexion committed:
Merge branch 'staging' into 10493-trialsessions

TomElliottFlexion committed:
10493: move eligible cases display limit from query to front end helper

TomElliottFlexion committed:
10493: WIP small fixes

zacharybrown204 committed:
Merge remote-tracking branch 'origin/staging' into 10493-trialsessions

TomElliottFlexion committed:
10493: missing test data

TomElliottFlexion committed:
Merge branch '10493-trialsessions' of https://github.com/ustaxcourt/ef-cms into 10493-trialsessions

jimlerza commented:
The prod deployment workflow, `build-and-deploy-to-prod`, runs automatically on every commit to prod.

If I'm understanding this correctly, we're going to have to cancel that build and manually run it again with the migration parameter set?

TomElliottFlexion commented:
Oh no, I definitely forgot that prod deploys automatically. I'll leave it up to you; we can cancel and set the param, we can just have it run without checking the param and remove the step before the _next_ deploy, we can run the script manually

TomElliottFlexion committed:
Merge branch 'staging' of https://github.com/ustaxcourt/ef-cms into 10493-trialsessions

zacharybrown204 committed:
8827 - Updating create trial session working

Update trial session working copy will now replace any existing records on key conflict. Though rare a judge could be removed and reassigned to a trial session. In this case we want to give them a blank working copy.

zacharybrown204 committed:
Merge branch 'staging' into 10493-trialsessions

jimlerza committed:
Merge branch 'staging' into 10493-trialsessions

TomElliottFlexion committed:
Merge branch 'staging' into 10493-trialsessions

zacharybrown204 committed:
8827 - Removing trial sessions todo file

jimlerza committed:
Merge branch 'staging' into 10493-trialsessions

TomElliottFlexion committed:
Merge branch 'staging' into 10493-trialsessions

jimlerza committed:
Merge branch 'staging' into 10493-trialsessions

jimlerza committed:
8827: perform migrate-trial-session-information in the migrate step

jimlerza committed:
Merge branch 'staging' into 10493-trialsessions
